### PR TITLE
chore: add experimental flag for docker compatibility

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
For being able to hide/show the new screen for compatibility mode

It brings a configuration flag. It is ignored for now (and hidden)

### What issues does this PR fix or reference?

https://github.com/containers/podman-desktop/issues/8337

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

**Review by deboer-tim — APPROVED** [review]
It's not totally clear to me how this setting would be used in the UI, but the change is small and no concerns.

File: packages/main/src/plugin/docker/docker-compatibility.spec.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+/* eslint-disable @typescript-eslint/no-empty-function */
+beforeAll(() => {
+  configurationRegistry = new ConfigurationRegistry({} as ApiSenderType, {} as Directories);
+  configurationRegistry.registerConfigurations = vi.fn();
+  configurationRegistry.deregisterConfigurations = vi.fn();
+});
+
+test('should register a configuration', async () => {
+  const appearanceInit = new DockerCompatibility(configurationRegistry);
```
Comment (jeffmaury) [inline packages/main/src/plugin/docker/docker-compatibility.spec.ts]: Maybe we should rename this variable for better readability

File: packages/main/src/plugin/docker/docker-compatibility.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  static readonly ENABLED_FULL_KEY = `${ExperimentalSettings.SectionName}.${ExperimentalSettings.Enabled}`;
+
+  #configurationRegistry: ConfigurationRegistry;
+
+  constructor(configurationRegistry: ConfigurationRegistry) {
+    this.#configurationRegistry = configurationRegistry;
+  }
+
+  init(): void {
+    const appearanceConfiguration: IConfigurationNode = {
```
Comment (jeffmaury) [inline packages/main/src/plugin/docker/docker-compatibility.ts]: Same here

Comment (benoitf) [reply packages/main/src/plugin/docker/docker-compatibility.ts]: fixed

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**